Kudu.Core/Jobs/JobsManagerBase.cs

```diff
@@ -100,7 +102,10 @@ public void RegisterExtraEventHandlerForFileChange(Action<string> action)
 
         private void OnJobChanged(string jobName)
         {
-            JobListCache = null;
+            lock (jobsListCacheLockObj)
```

Comment:
well the lock is needed because the nulling can happen during the collection of information about jobs list, if that happens we end up with an invalid object that gets assigned to the list and there is no way of knowing that an invalidation happened during information collection. 

the local variable is not needed at this point actually so will clean that up, will add a method to invalidate the cache. also locking should not be a big problem as it happens on a site level, so the number of listing and changing jobs is not expected to be huge.